test/integration/fuzzing_test.lua

```diff
@@ -0,0 +1,2687 @@
+            -- (17) Argument: [T!]!->Value: [] - OK
+            {{{'list', NonNullable, k, NonNullable, {}, nil, nil, nil, nil}}, nil},
+            -- (18) Argument: [T!]!->Value: [null] - FAIL
+            {
+                {{'list', NonNullable, k, NonNullable, {box.NULL}, nil, nil, nil, nil}},
+                'Expected non-null for \"NonNull('..v.var_type..')\", got null',
+            },
+            -- (19) Argument: [T!]!->Value: nil - FAIL
+            {
+                {{'list', NonNullable, k, NonNullable, nil, nil, nil, nil, nil}},
+                'Expected non-null for \"NonNull(List(NonNull('..v.var_type..')))\", got null',
+            },
+            -- (20) Argument: [T!]!->Value: null - FAIL
+            {
+                {{'list', NonNullable, k, NonNullable, box.NULL, nil, nil, nil, nil}},
+                'Expected non-null for \"NonNull(List(NonNull('..v.var_type..')))\", got null',
+            },
+        }
+        check_suite('Single list argument', test_suite)
+    end
+end
+
+-- Test single scalar, inputObject or enum argument and value provided with variable
+g.test_nonlist_arguments_with_variables_nullability = function ()
+    for k, v in pairs(graphql_types) do
+        local test_suite = {
+            -- (1) Argument: T->Variable: T->Value: value-> Default: value - OK
+            {{{k, Nullable, nil, nil, v.value, Nullable, nil, nil, v.default}}, nil},
+            -- (2) Argument: T->Variable: T->Value: value-> Default: nil - OK
+            {{{k, Nullable, nil, nil, v.value, Nullable, nil, nil, nil}}, nil},
+            -- (3) Argument: T->Variable: T->Value: value-> Default: null - OK
+            {{{k, Nullable, nil, nil, v.value, Nullable, nil, nil, box.NULL}}, nil},
+            -- (4) Argument: T->Variable: T->Value: nil-> Default: value - OK
+            {{{k, Nullable, nil, nil, nil, Nullable, nil, nil, v.default}}, nil},
+            -- (5) Argument: T->Variable: T->Value: nil-> Default: nil - OK
+            {{{k, Nullable, nil, nil, nil, Nullable, nil, nil, nil}}, nil},
+            -- (6) Argument: T->Variable: T->Value: nil-> Default: null - OK
+            {{{k, Nullable, nil, nil, nil, Nullable, nil, nil, box.NULL}}, nil},
+            -- (7) Argument: T->Variable: T->Value: null-> Default: value - OK
+            {{{k, Nullable, nil, nil, box.NULL, Nullable, nil, nil, v.default}}, nil},
+            -- (8) Argument: T->Variable: T->Value: null-> Default: nil - OK
+            {{{k, Nullable, nil, nil, box.NULL, Nullable, nil, nil, nil}}, nil},
+            -- (9) Argument: T->Variable: T->Value: null-> Default: null - OK
+            {{{k, Nullable, nil, nil, box.NULL, Nullable, nil, nil, box.NULL}}, nil},
+            -- (10) Argument: T->Variable: T!->Value: value-> Default: value - FAIL
+            {
+                {{k, Nullable, nil, nil, v.value, NonNullable, nil, nil, v.default}},
+                'Non-null variables can not have default values',
```

Comment:
This [issue](https://github.com/graphql/graphql-spec/issues/570) in graphql/graphql-spec repo states that they can. JS implementation does not return any errors.

Reply:
```javascript
var test = require('unit.js');
var { graphql, buildSchema } = require('graphql');

var rootValue = {
  test: (args) => {
    return args;
  },
}

function PrintResponse(response) {
  console.log(JSON.parse(JSON.stringify(response.data)));
}

var schema = buildSchema(`
  type result {
    arg1: Float
  }

  type Query {
    test(arg1: Float): result
  }
`)

var query = `query MyQuery($var1: Float! = 0) { test(arg1: $var1) { arg1 } }`

graphql({
  schema,
  source: query,
  rootValue
}).then(PrintResponse)

graphql({
  schema,
  source: query,
  rootValue,
  variableValues: { var1: 1.12}
}).then(PrintResponse)
```

```js
{ test: { arg1: 0 } }
{ test: { arg1: 1.12 } }
```